From 39ba122599133b7a1dc81439109ecf59d1f27b1f Mon Sep 17 00:00:00 2001
From: Vitaly Andrianov <vitalya@ti.com>
Date: Fri, 12 May 2017 09:53:43 -0400
Subject: [PATCH 27/28] arm: dts: am572x: add am572x-idk-jailhouse dts

This commit add the am572x-idk-jailhouse.dts which is extension of the
am572x-idk.dts. It allows to demonstrate Jailhouse hypervisor and run
couple of bare-metal as well as TI-RTOS test applications like PRUSS and
ICSS_EMAC.

1) reserve memory for jailhouse and jailhouse inmate
2) disable timer2 and timer8 at Linux but left them power-on
3) disable uart9 (with some evm modification uart9 may be used by an
   inmate)
4) disable pruss, gpio3 and gpio5. They will be used by TI-RTOS examples.
5) skip some interrupts to be used by inmates

Signed-off-by: Vitaly Andrianov <vitalya@ti.com>
Signed-off-by: Jacob Stiffler <j-stiffler@ti.com>
---
 arch/arm/boot/dts/am572x-idk-jailhouse.dts | 106 +++++++++++++++++++++++++++++
 1 file changed, 106 insertions(+)
 create mode 100644 arch/arm/boot/dts/am572x-idk-jailhouse.dts

diff --git a/arch/arm/boot/dts/am572x-idk-jailhouse.dts b/arch/arm/boot/dts/am572x-idk-jailhouse.dts
new file mode 100644
index 0000000..ed08fb9
--- /dev/null
+++ b/arch/arm/boot/dts/am572x-idk-jailhouse.dts
@@ -0,0 +1,106 @@
+/*
+ * Copyright (C) 2017 Texas Instruments Incorporated - http://www.ti.com/
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "am572x-idk.dts"
+
+/ {
+	reserved-memory {
+		jailhouse: jailhouse@ef000000 {
+			reg = <0x0 0xef000000 0x0 0x1000000>;
+			no-map;
+			status = "okay";
+		};
+
+		jh_inmate: jh_inmate@ee000000 {
+			reg = <0x0 0xe0000000 0x0 0xf000000>;
+			no-map;
+			status = "okay";
+		};
+	};
+};
+
+&timer2 {
+	status = "disabled";
+	ti,no-idle;
+};
+
+&timer8 {
+	status = "disabled";
+	ti,no-idle;
+};
+
+&uart9 {
+	status = "disabled";
+	ti,no-idle;
+};
+
+/ {
+	ocp {
+		pruss1_eth {
+			status = "disabled";
+		};
+
+		pruss2_eth {
+			status = "disabled";
+		};
+
+		crossbar_mpu: crossbar@4a002a48 {
+			ti,irqs-skip = <10 44 127 129 133 134 135 136 137 139 140>;
+		};
+	};
+};
+
+&pruss1 {
+	status = "disabled";
+	ti,no-idle;
+};
+
+&pru1_0 {
+	status = "disabled";
+	ti,no-idle;
+};
+
+&pru1_1 {
+	status = "disabled";
+	ti,no-idle;
+};
+
+&pruss2 {
+	status = "disabled";
+	ti,no-idle;
+};
+
+&pru2_0 {
+	status = "disabled";
+	ti,no-idle;
+};
+
+&pru2_1 {
+	ti,no-idle;
+	status = "disabled";
+};
+
+&pruss1_mdio {
+	ti,no-idle;
+	status = "disabled";
+};
+
+&pruss2_mdio {
+	ti,no-idle;
+	status = "disabled";
+};
+
+&gpio5 {
+	ti,no-idle;
+	status = "disabled";
+};
+
+&gpio3 {
+	ti,no-idle;
+	status = "disabled";
+};
-- 
1.9.1

